pkg/controller/provider/container/openstack/model.go

```diff
@@ -258,14 +260,15 @@ func (r *ImageAdapter) List(ctx *Context) (itr fb.Iterator, err error) {
 		list.Append(m)
 	}
 	itr = list.Iter()
-
+	r.lastSync = now
 	return
 }
 
-func (r *ImageAdapter) GetUpdates(ctx *Context, lastSync time.Time) (updates []Updater, err error) {
+func (r *ImageAdapter) GetUpdates(ctx *Context) (updates []Updater, err error) {
 	opts := &ImageListOpts{}
-	opts.setUpdateAtQueryFilterGTE(lastSync)
+	opts.setUpdateAtQueryFilterGTE(r.lastSync)
 	imageList := []Image{}
+	now := time.Now()
```

Comment:
I think we still have a small race between lines 269-271, maybe it would make sense to move it to line 268?

Reply:
yeah, it's better to be on the "safe side" and initiate a redundant update later on than missing an update

Reply:
Sorry but I don't follow, `time.Now()` is taken before doing the actual query, probably some microseconds before. So  if an update is done while updating the `Images`, it will be refreshed in the next iteration, right?

Reply:
ack, I read it as if setUpdateAtQueryFilterGTE triggers the call